From 258e3065687c9e6b53b6f2b6c7fa6c5694508791 Mon Sep 17 00:00:00 2001
From: Lubos Kardos <lkardos@redhat.com>
Date: Mon, 23 May 2016 10:19:39 +0200
Subject: [PATCH] Fix signing with non-ASCII uid keys (rhbz:1243963)

Removed setting LC_ALL to "C" because since commit [1] the gpg program
gets password by yourself from terminal so there is no sense in
setting LC_ALL to "C" if the terminal settings is e. g. UTF-8. That was
only confusing gpg program and it was not able to properly get and
display non-ASCII characters.

[1] 0bce5fcf270711a2e077fba0fb7c5979ea007eb5
---
 sign/rpmgensig.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/sign/rpmgensig.c b/sign/rpmgensig.c
index 2310f14..321d19c 100644
--- a/sign/rpmgensig.c
+++ b/sign/rpmgensig.c
@@ -267,7 +267,6 @@ static int runGPG(sigTarget sigt, const char *sigfile)
 
 	if (gpg_path && *gpg_path != '\0')
 	    (void) setenv("GNUPGHOME", gpg_path, 1);
-	(void) setenv("LC_ALL", "C", 1);
 
 	unsetenv("MALLOC_CHECK_");
 	cmd = rpmExpand("%{?__gpg_sign_cmd}", NULL);
-- 
1.9.3

